dist/controller.py

```diff
@@ -0,0 +1,75 @@
+import os
+import subprocess
+import argparse
+import sys
+import re
+
+
+def extract_args():
```

Comment:
Type hints and proper method / script documentation has to be added to the whole `controller.py`.

Reply:
Please address this in the SPIKE issue as one of the goals.